storage/fanout.go

```diff
@@ -569,11 +593,14 @@ func (h *genericSeriesSetHeap) Pop() interface{} {
 	return x
 }
 
-// ChainedSeriesMerge returns single series from many same series by chaining samples together.
+// ChainingSeriesMerge returns single series from many same series by chaining samples together.
 // In case of the timestamp overlap, the first overlapped sample is kept and the rest samples with the same timestamps
```

Comment:
> the first overlapped sample is kept

I just noticed this. In that case, we need to be consistent (maybe a future PR for someone). When vertical queries and compaction was introduced, the design was to keep the last appearing sample.

Reply:
Will check thanks!

Reply:
I mean... what is last and first? It's really random and always was... which one is first which last ;p  Maybe not from algo perspective but from the caller perspective - we randomly put results from different remote APIs (in random order)

Reply:
Changed docs to have "random one"

Reply:
If we sort the blocks based on their minTime, then the block with lower mint is first and higher is the last - that was what was used in compaction and vertical query. Because we use heap, I don't think that is easy to take care of it.